File: medium/875-Koko-Eating-Bananas/BinarySearch.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        left = 0
        right = max(piles)
        
        while right - left > 1:
            mid = (left + right) // 2
            required_hour = sum((p + mid - 1) // mid for p in piles)
            if required_hour <= h:
                right = mid
            else:
                left = mid
        
        return right

File: medium/875-Koko-Eating-Bananas/BinarySearch.kt
class Solution {
    fun minEatingSpeed(piles: IntArray, h: Int): Int {
        var low = 0
        var high = piles.max()!!
        
        while (high - low > 1) {
            val mid = (high + low) / 2
            var requiredHour: Long = 0
            for (pile in piles) {
                requiredHour += (pile + mid - 1) / mid
            }
            if (requiredHour <= h) {
                high = mid
            } else {
                low = mid
            }
        }
        
        return high
    }
}
